_posts/2022-08-29-changelog-177.markdown

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #177"
+date: 2022-08-29 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#9200] Provide debugger pretty-printers to render `alloc::rc::Weak` and `alloc::sync::Weak`
+
+* [#9168] Add debugger pretty-printers to show the contents and the size of Rust slices
+
+* [#9156] Improve top-level completion of macro calls
+
+* [#9137] extended list of key completions in `Cargo.toml` (by [@afetisov])
+
+* [#9006] support rename refactoring of items expanded from a macro
+
+* [#8937] Enables unused imports inspection if proc macros are enabled. Proc macros can be enabled using `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental features](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
+
+## Performance Improvements
+
+* [#9208] Slightly speed up name resolution
+
+* [#9190] Slightly speed up name resolution
+
+* [#9182] Slightly speed up name resolution
+
+* [#9161] Slightly speed up type inference
+
+* [#9104] Slightly speed up name resolution
+
+## Fixes
+
+* [#9203] Update debugger pretty-printers to render `Ref` and `RefMut` properly
+
+* [#9194] Fixes possible `java.io.IOException` during typing in a macro call
+
+* [#9183] COMP: Fix duplicate top-level completion of macro calls
+
+* [#9175] Fix proc macro expansion on Rust 1.63 and 1.64 nightly
+
+* [#9173] MACRO: Use `rust-analyzer-proc-macro-srv` from toolchain if available
```

Comment:
Adopted [rust-analyzer's changelog](https://rust-analyzer.github.io/thisweek/2022/08/01/changelog-140.html):

```suggestion
* [#9173] Rustup toolchains newer than 2022-07-29 contain a proc macro server that should be compatible with macros built by the matching compiler. Since today’s release, the server is automatically detected by and used the plugin. This means that proc macros will keep working on latest Rust nightly versions. For more details, you can read @fasterthanlime’s [blog post](https://fasterthanli.me/articles/proc-macro-support-in-rust-analyzer-for-nightly-rustc-versions#what-s-next).
```